### ISSUE TITLE
Add a "force-delete" to history --delete

### ISSUE THREAD

**Issue by Selora** [fish-shell/fish-shell#3843]
**fish version installed **:
 fish, version 2.5.0

Currently, the only way to remove history entries non interactively is with an exact match. It would be nice to add a --force to flush all entries with the 'prefix' or 'contains' flag.

**Comment by krader1961** [comment]
What we actually want to do is augment the builtin so that you can pass a non-exact match. That is, it should differ from the function only in that the function will ask for confirmation. The `history` function should continue to require the user to review the matches and enter `all` or the IDs of the items to be deleted. We do not want a `--force` flag.

I thought we had an open issue to do that but can't find it so we'll use this to track making that change. Feel free to take a stab at enhancing the `history delete` builtin.

**Comment by xpe** [comment]
I recommend closing this issue. As of version 4.9.2:

`> history delete -` <kbd>tab</kbd>

(autocomplete...)

```console
-C  --case-sensitive  (Match items in a case-sensitive manner)
-c  --contains        (Match items containing the string)
-e  --exact           (Match items identical to the string)
-h  --help            (Display help and exit)
-n  --max             (Limit output to the first 'n' matches)
-p  --prefix          (Match items beginning with the string)
-R  --reverse         (Output the oldest results first)
-t  --show-time       (Output with timestamps)
-z  --null            (Terminate entries with NUL character)
--color               (When to colorize output)
```

**Comment by krobelus** [comment]
No, this issue is about `builtin history delete`, not the `history` function that wraps it.
